docs/boilerplate.md

```diff
@@ -0,0 +1,126 @@
+├── app.ts
+├── index.html
+├── package.json
+├── (server/)
+│   ├──render.ts
+│   └──server.ts
+```
+
+Here's a breakdown of the different bits of boilerplate code:
+
+### `package.json`
+
+The package.json file contains all the dependencies of a node js project. To simplify the dependency management, all dependencies are pulled through a single preset package. The preset package contains dependencies to all available oryx packages. This might not be the most optimal setup for your project over time, as it might contain a lot of "dead code", but for the start of your project it is very convenient. While you're getting experienced with Oryx over time, you could consider to build your own list of dependencies to get more fine-grained control. That being said, having unused dependencies in your project will not affect the build time or run time of your project. It is only an overhead during the npm install process.
+
+The bare minimum package.json is setup with the following package.json:
+
+```json
+{
+  "name": "my-oryx-app",
+  "dependencies": {
+    "@spryker-oryx/presets": "^1.0.0"
+  },
+  "devDependencies": {
+    "vite": "^4.0.0"
+  }
+}
+```
+
+### `index.html`
+
+The `index.html` file contains two lines of boilerplate code to bootstrap the application:
+
+1. the `root-app` element
+2. an import of the `app.ts` module
+
+Both the `root-app` element and the `app.ts` module can be replaced with custom alternatives.
+
+While the index.html can have a few more details, the bare minimum required code is shown in the snippet below:
+
+```html
+<html>
+  <body>
+    <root-app></root-app>
+  </body>
+  <script type="module" src="./app.ts"></script>
+</html>
+```
+
+Oryx allows to be installed and used next to other application code. You could therefor have other elements and code that are not related to Oryx. Oryx is bootstrapped with the `oryx-app` root element by default, but you can set it up to use an alternative root element.
+
+### `app.ts`
+
+The `app.ts` file contains the bootstrap code of the application. Bootstrapping the application can be done with a the `appBuilder`, a function that allows to configure the application. While the `appBuilder` can be configured with a lot of fine-grained configurations, the bare minimum setup is very clean:
+
+```ts
+import { appBuilder } from '@spryker-oryx/core';
+import { b2cFeatures } from '@spryker-oryx/presets';
+import { storefrontTheme } from '@spryker-oryx/themes';
+
+export const app = appBuilder()
+  .withFeature(b2cFeatures)
+  .withTheme(storefrontTheme)
+  .withEnvironment(import.meta.env)
+  .create();
+```
+
+In the boilerplate we see the following things:
+
+- the usage of a standard _feature set_, which is provided by the `presets` package
+- the usage of a standard theme, that applies a global color scheme and other theme design tokens to the application
+- the usage of environment variables, to provide the backend base URLs to the application
+
+### `server/render.ts` (optional)
```

Comment:
Hm, we are describing the file but not even showing the content, what is the purpose of having it?